File: Chapter.js
class Chapter extends TableOfContents {
  constructor(name) {
    super();
    this.name = name;
    this.subChapters = [];
  }

  addSubChapter(name) {
    const SubChapter = require('./Subchapter');
    const sc = new SubChapter(name);
    this.subChapters.push(sc);
    return sc;
  }

}

File: src/Chapter.java
import java.util.ArrayList;
import java.util.List;

public class Chapter extends TableOfContents {
    String name;
    private final List<SubChapter> subChapters = new ArrayList<>();

    public Chapter(String name) {
        this.name = name;
    }

    public SubChapter addSubChapter(String name) {
        SubChapter sc = new SubChapter(name);
        subChapters.add(sc);
        return sc;
    }

    public List<SubChapter> getSubChapters() {
        return subChapters;
    }
}

